src/toxcore/messenger/packet/user_status.rs

```diff
@@ -0,0 +1,81 @@
+/*! UserStatus struct.
+*/
+
+use nom::le_u8;
+
+use crate::toxcore::binary_io::*;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[repr(u8)]
+/// Status of me
+pub enum PeerStatus {
+    /// Online
+    Online = 0,
+    /// Away
+    Away,
+    /// Online but I am busy
+    Busy,
+}
+
+impl FromBytes for PeerStatus {
+    named!(from_bytes<PeerStatus>,
+        switch!(le_u8,
+            0 => value!(PeerStatus::Online) |
+            1 => value!(PeerStatus::Away) |
+            2 => value!(PeerStatus::Busy)
+        )
+    );
+}
+
+/** UserStatus is a struct that holds status of me.
+
+This packet is used to transmit sender's status to a friend.
+Every time a friend become online or my status is changed,
+this packet is sent to the friend or to all friends of mine.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x32`
+`1`       | My status(0 = online, 1 = away, 2 = busy)
+
+*/
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct UserStatus {
```

Comment:
Why don't you make UserStatus be enum?

```
pub enum UserStatus {
    /// Online
    Online = 0,
    /// Away
    Away,
    /// Online but I am busy
    Busy,
}
```

Reply:
What if we need to use status somewhere else? For example, state saving/loading.

Reply:
Yes but business logic or state should not be described in /packets/ module. We'd better duplicate it somewhere else

Reply:
Why duplicate when we can move it to a common part?

Reply:
And it seems it's actually saved: https://github.com/TokTok/c-toxcore/blob/0aad180d1e1d7dae7cbf2868e139af0a4dc34fe7/toxcore/Messenger.c#L2812

Reply:
Also we will need to store friend's status in memory and return it to a client. So probably it can be stored somewhere in `toxcore/friends`. But anyway we can put it later to appropriate place, I think.

Reply:
There should be different structs to store inside Saved_Friend and to send through the network.

Reply:
done